packages/tx/src/baseTransaction.ts

```diff
@@ -91,8 +91,20 @@ export abstract class BaseTransaction<TransactionObject> {
     const rB = toBuffer(r === '' ? '0x' : r)
     const sB = toBuffer(s === '' ? '0x' : s)
 
-    this.nonce = new BN(toBuffer(nonce === '' ? '0x' : nonce))
-    this.gasLimit = new BN(toBuffer(gasLimit === '' ? '0x' : gasLimit))
+    const nonceB = toBuffer(nonce === '' ? '0x' : nonce)
+    if (nonceB.length > 8) {
+      // Implements EIP-2681 which limits nonce to 2^64-1
+      // https://eips.ethereum.org/EIPS/eip-2681
+      throw new Error('nonce cannot be more than 2^64-1')
+    }
+    this.nonce = new BN(nonceB)
+
+    const gasLimitB = toBuffer(nonce === '' ? '0x' : gasLimit)
+    if (gasLimitB.length > 8) {
+      throw new Error('gasLimit cannot be more than 2^64-1')
+    }
+    this.gasLimit = new BN(gasLimitB)
```

Comment:
This is all Dmitry could tell me:
>I think since the block gas limit is capped by eip so the transaction gas limit

[These tests](https://github.com/ethereum/tests/blob/develop/TransactionTests/ttGasLimit/TransactionWithGasLimitOverflow64.json) fail if this limit is not in place.

Reply:
That said, Geth limits gasLimit to a [Uint64](https://github.com/ethereum/go-ethereum/blob/33f28138095faada5cbf027849993808a9dba785/core/genesis.go#L54) so this check would apply the same since we don't have an exact Uint64 in Javascript.

Reply:
Ok, thanks, maybe just add "Geth limits gasLimit to a Uint64" as a one-line comment above here.